### PR TITLE
ENGINES: Implement heartbeat, use native KotOR AI

### PR DESCRIPTION
**Pull request by seedhartha**
I turns out that AI in KotOR games is handled by a special script (which is open source, BTW). I implemented the heartbeat event to invoke that script, as well as some of the missing functions to make Trask follow the PC.

### CODE REVIEW

File: src/engines/kotorbase/module.cpp [hunk 1]
```diff
@@ -673,6 +677,17 @@ void Module::handleActions() {
 	}
 }
 
+void Module::handleHeartbeat() {
+	const int kHeartbeatInterval = 1000; // ms
```
Comment (DrMcCoy) [inline src/engines/kotorbase/module.cpp]: Is every second correcct for KotOR? In NWN, the heartbeat is invoked every 6 seconds (= the length of a combat round)

Comment (seedhartha) [reply src/engines/kotorbase/module.cpp]: I'm not really sure. What would be the right place to look for a reference?

Comment (DrMcCoy) [reply src/engines/kotorbase/module.cpp]: On Deadly Stream, a KotOR modding site, they also say 6s: https://deadlystream.com/topic/6371-on-heartbeat/ (the Lexicon is https://nwnlexicon.com/) , https://deadlystream.com/topic/2062-tsl-textures-remastered/?page=9&tab=comments#comment-23797

Comment (DrMcCoy) [reply src/engines/kotorbase/module.cpp]: In the future, we should probably have a "round handler" or something in the KotOR engine, that synchronously does all the things required per round. I.e. calling all heartbeat scripts, evaluating pathfinding, checking and firing script events like perception, going through combat initiative, etc.

Comment (seedhartha) [reply src/engines/kotorbase/module.cpp]: Changed the interval to 6 seconds.

**Comment by DrMcCoy** [comment]
Merged as ef40916e2...82273fc5f, thanks! :)

**Pull request closed without merging**